operator/src/lib.rs

```diff
@@ -235,36 +162,152 @@ impl OpaState {
-                            node_type,
+                            role_str,
                             role_group
                         );
 
-                        // Create a pod for this node, role and group combination
-                        let pod = build_pod(
-                            &self.context.resource,
-                            node_name,
-                            &labels,
-                            &pod_name,
-                            &cm_name,
-                            opa_config,
-                        )?;
+                        let (pod, config_maps) = self
+                            .create_pod_and_config_maps(
+                                &role,
+                                role_group,
+                                &node_name,
+                                config_for_role_and_group(
+                                    role_str,
+                                    role_group,
+                                    &self.validated_role_config,
+                                )?,
+                            )
+                            .await?;
+
                         self.context.client.create(&pod).await?;
+
+                        for config_map in config_maps {
+                            self.create_config_map(config_map).await?;
+                        }
                     }
                 }
             }
         }
         Ok(ReconcileFunctionAction::Continue)
     }
+
+    async fn create_pod_and_config_maps(
+        &self,
+        role: &OpaRole,
+        role_group: &str,
+        node_name: &str,
+        validated_config: &HashMap<PropertyNameKind, BTreeMap<String, String>>,
+    ) -> Result<(Pod, Vec<ConfigMap>), Error> {
+        let mut config_maps = vec![];
+        let mut env_vars = vec![];
+        let mut start_command = vec![];
+
+        let mut cm_data = BTreeMap::new();
+        let pod_name = build_pod_name(
+            APP_NAME,
+            &self.context.name(),
+            &role.to_string(),
+            role_group,
+            node_name,
+        );
+        let cm_name = format!("{}-config", pod_name);
+
+        for (property_name_kind, config) in validated_config {
+            match property_name_kind {
+                PropertyNameKind::File(file_name) => {
+                    if file_name.as_str() == CONFIG_FILE {
+                        if let Some(repo_reference) = config.get(REPO_RULE_REFERENCE) {
+                            cm_data.insert(
+                                CONFIG_FILE.to_string(),
+                                create_config_file(repo_reference),
+                            );
+                            config_maps.push(
+                                ConfigMapBuilder::new()
+                                    .metadata(
+                                        ObjectMetaBuilder::new()
+                                            .name(cm_name.clone())
```

Comment:
```suggestion
                                            .name(&cm_name)
```

Reply:
Its used in line 287:
`container_builder.add_configmapvolume(cm_name, "conf".to_string());`